lib/puppetserver/ca/action/prune.rb

```diff
@@ -50,26 +55,39 @@ def run(inputs)
 
           # Getting the CRL(s)
           loader = X509Loader.new(puppet.settings[:cacert], puppet.settings[:cakey], puppet.settings[:cacrl])
+          inventory_file = puppet.settings[:cert_inventory]
+          cadir = puppet.settings[:cadir]
 
           verified_crls = loader.crls.select { |crl| crl.verify(loader.key) }
+          number_of_removed_duplicates = 0
+          number_of_removed_crl_entries = 0
 
           if verified_crls.length == 1
             puppet_crl = verified_crls.first
             @logger.inform("Total number of certificates found in Puppet's CRL is: #{puppet_crl.revoked.length}.")
-            number_of_removed_duplicates = prune_CRL(puppet_crl)
 
-            if number_of_removed_duplicates > 0
+            if serialnumbers
+              number_of_removed_crl_entries += prune_using_serial(puppet_crl, loader.key, serialnumbers)
+            end
+            if certnames
+              number_of_removed_crl_entries += prune_using_certname(puppet_crl, loader.key, inventory_file, cadir, certnames)
+            end
+            if (!serialnumbers && !certnames)
+               number_of_removed_duplicates += prune_CRL(puppet_crl)
```

Comment:
Should the `number_of_removed_crl_entries` contain the sum of the `serial`, `certname` and `duplicate` matches?  I got the impression that was the intent.